src/components/EventsTimeline/components/EventData.tsx

```diff
@@ -47,7 +47,14 @@ const EventData: React.FC<Props> = ({ fields, values, isNested = false }) => {
 
   const displayValues =  fields.reduce((acc, { key, label, shouldShow, mapValue, italic, isCurrency }) => {
     const value = values[key]
-    if (value == null || shouldShow(value, isNested) === false) return acc
+    /**
+     * don't show label and value when
+     * - value == null
+     * - fn shouldshow returns false
+     * - value is an array and this array is empty, 
+     */
+    
+    if (value == null || shouldShow(value, isNested) === false || (Array.isArray(value) && value.length === 0)) return acc
```

Comment:
`shouldShow(value, isNested) === false`

or

`!shouldShow(value, isNested)`

Reply:
I think this is doing the same thing.
`if (!(value.length > 0) || !shouldShow(value, isNested)) return acc`

Reply:
The type of `value` is unknown and can be anything, so we can't check for length here